xcodecache/swiftpm_cache.go

```diff
@@ -0,0 +1,28 @@
+package cache
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/bitrise-io/go-steputils/cache"
+)
+
+// CollectPackagesCache marks the Swift Package Manager packages to be added the cache
+// The directory cached is: $HOME/Library/Developer/Xcode/DerivedData/[PER_PROJECT_DERIVED_DATA]/SourcePackages
+func CollectPackagesCache(projectPath string) error {
+	projectDerivedData, err := xcodeProjectDerivedDataPath(projectPath)
+	if err != nil {
+		return fmt.Errorf("%s", err)
+	}
+
+	projectSwiftPMDir := path.Join(projectDerivedData, "SourcePackages")
+
+	cache := cache.New()
+	cache.IncludePath(projectSwiftPMDir)
```

Comment:
is not there any marker file, which could invalidate the cache in case of change?

Reply:
There is a dependencies-state.json that could be used, I did not fully check that it always changes if the contents/state change tough.